Mathlib/Tactic/Linter/FlexibleLinter.lean

```diff
@@ -389,17 +388,14 @@ def flexibleLinter : Linter where run := withSetOptionIn fun _stx => do
     for d in getStained! s do
       if shouldStain? then
         for currMVar1 in mvs1 do
-          let lctx1 := ((ctx1.decls.find? currMVar1).getD default).lctx
+          let lctx1 := (ctx1.decls.findD currMVar1 default).lctx
           let locsAfter := d.toFMVarId currMVar1 lctx1
-
-          for l in locsAfter do
-            stains := stains.push (l, (d, s))
-
+          stains := stains.append (locsAfter.map (fun l ↦ (l, (d, s))))
```

Comment:
This replaces many `push` operations by one `append` (whose capacity is predictable): hopefully, that's a tad more efficient.

Reply:
I think it's not actually more efficient, `append` doesn't use the known capacity at present. But may it will in future, and in any case the code is clearer!

Please use `++` instead of append, though.

Reply:
Actually: one *could* do this by
- adding another parameter `min_cap` to `lean_array_push`
- in that function, ensure the new capacity is at least `min_cap` --- something like this, perhaps better optimised
```spoiler
extern "C" LEAN_EXPORT object * lean_array_push(obj_arg a, obj_arg v, size_t min_cap) {
    object * r;
    if (lean_is_exclusive(a)) {
        if (lean_array_capacity(a) > std::max(lean_array_size(a), min_cap))
            r = a;
        else
            r = lean_copy_expand_array(a, true);
    } else {
        r = lean_copy_expand_array(a, min_cap < lean_array_capacity(a) && lean_array_capacity(a) < 2*lean_array_size(a) + 1);
    }
    lean_assert(lean_array_capacity(r) > lean_array_size(r));
    size_t & sz  = lean_to_array(r)->m_size;
    object ** it = lean_array_cptr(r) + sz;
    *it = v;
    sz++;
    return r;
}
```
- `Array.append` would need to be modified. Perhaps `Array.push` can internally also take a `minCapacity` parameter, which this would call? (And I'm not sure what the ramifications of this change are, if e.g. other interpreted code would get slower...)

In any case, I'm not sure if this is the right level at which to fix this, or one should do this deeper in the call stack. Would opening an issue about this be useful/is this known already?